app/src/androidTest/java/com/github/se/project/ui/EndToEnd.kt

```diff
@@ -0,0 +1,159 @@
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.uiautomator.UiDevice
+import com.github.se.project.PocketTutorApp
+import com.github.se.project.model.lesson.Lesson
+import com.github.se.project.model.lesson.LessonRepository
+import com.github.se.project.model.lesson.LessonViewModel
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.ProfilesRepository
+import com.github.se.project.ui.navigation.NavigationActions
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.any
+import org.mockito.kotlin.whenever
+
+@RunWith(AndroidJUnit4::class)
+class EndToEndTest {
+
+  @Mock lateinit var navigationActions: NavigationActions
+
+  @Mock lateinit var context: Context
+
+  // Mock du ProfilesRepository
+  private val mockProfileRepository = Mockito.mock(ProfilesRepository::class.java)
+
+  private val mockProfileViewModel = ListProfilesViewModel(mockProfileRepository)
+
+  private val mockLessonRepository = mock(LessonRepository::class.java)
+
+  private val mockLessonViewModel = LessonViewModel(mockLessonRepository)
+
+  var currentLesson: Lesson? = null
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.addLesson(any(), any(), any())).thenAnswer { invocation ->
+      currentLesson = invocation.arguments[0] as Lesson
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    /*whenever(mockLessonRepository.getLessonsForStudent(any(), any(), any())).thenAnswer { invocation
+      ->
+      val onComplete = invocation.arguments[1] as (List<Lesson>) -> Unit
+      if (currentLesson != null) {
+        onComplete(listOf(currentLesson!!)) // Simulate a successful update
+      }
+    }*/
+  }
+
+  // End to end test, for the whole app, firebase included
+  @Test
+  fun EndToEnd() {
+    // Start the app in test mode
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    Thread.sleep(30000)
+
+    // Espresso.onView(isRoot()).check(matches(isDisplayed()))
+
+    // Sign in
+    composeTestRule.onNodeWithTag("loginButton").performClick()
+
+    // Enter valid data for all fields
```

Comment:
One thing that could be great to add is to test directly that the navigation is correct either by verifying that the navigateTo function is correclty called or at least by checking is the screen displayed is the correct one (using the tag on the scaffold of the screen we want to be displayed). It seems not to be a major check but since we have many screens that looks a bit the same, I think it is more secure to test directly the navigation as well in the E2E test.